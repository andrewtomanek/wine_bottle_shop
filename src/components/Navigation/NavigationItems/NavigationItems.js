import React from "react";
import { reduceCartState } from "../../../shared/helpers";
import classes from "./NavigationItems.module.css";
import NavigationItem from "./NavigationItem/NavigationItem";
import CartDrawer from "../CartToolbar/CartDrawer";

const NavigationItems = (props) => {
  const calculateCounter = (inventory) => {
    if (reduceCartState(inventory) === 0) return null;
    return (
      <span
        onClick={() => props.showCartDrawer(true)}
        onMouseEnter={() => props.showCartDrawer(true)}
        /* onMouseLeave={() => setIsCartVisible(false)} */
        className={classes.CartCounter}
      >
        {reduceCartState(inventory)}
      </span>
    );
  };

  return (
    <ul className={classes.NavigationItems}>
      <NavigationItem link="/" exact>
        Obchod
      </NavigationItem>
      {props.cartContent && props.isCartDrawerOpen && (
        <CartDrawer
          inventory={props.cartContent}
          listItems={props.listItems}
          price={props.price}
        />
      )}
      {props.cartContent && calculateCounter(props.cartContent)}
      {props.isAuthenticated ? (
        <NavigationItem link="/orders">Objednávky</NavigationItem>
      ) : null}
      {!props.isAuthenticated ? (
        <NavigationItem link="/auth">Registrace</NavigationItem>
      ) : (
        <NavigationItem link="/logout">Odhlásit</NavigationItem>
      )}
    </ul>
  );
};

export default NavigationItems;
