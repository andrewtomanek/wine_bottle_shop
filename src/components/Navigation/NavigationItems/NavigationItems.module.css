.NavigationItems {
  display: grid;
  grid-gap: 4rem;
  grid-auto-flow: row;
  margin: 0;
  font-family: 'Alegreya SC',monospace;
}

@media (min-width: 500px) {
  .NavigationItems {
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    align-items: center;
    height: 100%;
    flex-flow: row;
  }
}
