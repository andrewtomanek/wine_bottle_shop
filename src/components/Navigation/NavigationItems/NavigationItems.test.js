import React from "react";
import { shallow } from "enzyme";
import NavigationItems from "./NavigationItems";

const props = {
  isAuthenticated: true,
};

const falseProps = {
  isAuthenticated: false,
};

describe("Renders NavigationItems", () => {
  it("renders homepage link", () => {
    const wrapper = shallow(<NavigationItems {...props} />);
    expect(wrapper.find({ link: "/" }).text()).toEqual("<NavigationItem />");
  });

  it("renders orders link", () => {
    const wrapper = shallow(<NavigationItems {...props} />);
    expect(wrapper.find({ link: "/orders" }).text()).toEqual(
      "<NavigationItem />"
    );
  });

  it("hides orders link", () => {
    const wrapper = shallow(<NavigationItems {...falseProps} />);
    expect(wrapper.find({ link: "/orders" })).toEqual({});
  });

  it("renders logout link", () => {
    const wrapper = shallow(<NavigationItems {...props} />);
    expect(wrapper.find({ link: "/logout" }).text()).toEqual(
      "<NavigationItem />"
    );
  });

  it("renders NavigationItems component", () => {
    const wrapper = shallow(<NavigationItems {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
