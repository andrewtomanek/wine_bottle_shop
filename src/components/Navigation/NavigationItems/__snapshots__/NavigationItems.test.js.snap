// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NavigationItems renders NavigationItems component 1`] = `
<ul
  className="NavigationItems"
>
  <NavigationItem
    exact={true}
    link="/"
  >
    Obchod
  </NavigationItem>
  <NavigationItem
    link="/orders"
  >
    Objednávky
  </NavigationItem>
  <NavigationItem
    link="/logout"
  >
    Odhlásit
  </NavigationItem>
</ul>
`;
